Source/Blazorise/Components/Carousel/Carousel.razor.cs

```diff
@@ -269,79 +257,71 @@ private async void OnLocalizationChanged( object sender, EventArgs eventArgs )
             await InvokeAsync( StateHasChanged );
         }
 
-        private void InitializeTimer()
+        private Task InitializeTimer( double? interval = null )
         {
-            Timer = new( Interval );
-            Timer.Elapsed += OnTimerEvent;
-            Timer.AutoReset = true;
-        }
+            Timer = new( TimeSpan.FromMilliseconds( interval ?? Interval ) );
 
-        private void InitializeTransitionTimer()
-        {
-            TransitionTimer = new( 2000 );
-            TransitionTimer.Elapsed += OnTransitionTimerEvent;
-            TransitionTimer.AutoReset = false;
+            Task.Run( async () =>
+            {
+                while ( await Timer.WaitForNextTickAsync() )
+                {
+                    if ( AutoPlayEnabled )
+                        await OnTimerEvent( this, null );
+                };
+            } );
+            return Task.CompletedTask;
         }
 
-        private void SetTimer()
+        private Task InitializeTransitionTimer()
         {
-            TimerEnabled = ( Interval > 0 );
+            TransitionTimer = new( TimeSpan.FromMilliseconds( 2000 ) );
 
-            if ( Timer == null && TimerEnabled )
+            Task.Run( async () =>
             {
-                InitializeTimer();
-            }
-
-            if ( AutoPlayEnabled )
-            {
-                Timer.Start();
-            }
+                while ( await TransitionTimer.WaitForNextTickAsync() )
+                {
+                    //await OnTransitionTimerEvent( this, null );
```

Comment:
This is commented on purpose?

Reply:
Yea on purpose I replaced the transition timer calbsck with the new call on run animations... So basically. It seems we don't need transition timer at all but I'm. Not sure...